### DISCUSSION TITLE
Different rendering modes in the same codebase

### DISCUSSION THREAD

**Discussion by lucasdavidferrero** [nuxt/nuxt#29285] [Questions]
Hello community, currently I have 2 frontends. One is a Nuxt3 application, we're using universal rendering and ssr since it's an ecommer website that our clients use to purchase products. 

On the other hand, we have a Vue SPA Frontend that is used internally in the company for administrate the ecommerce website, also, it has modules for the company's business where the Nuxt is not aware of.

There's only one backend for both client apps.

We're thinking if is best to migrate the SPA to Nuxt. As a developer, this will be an avantage because everything is in one place, this allows us to re use code more easily. But in the other hand, I wonder if our customer experenci will be affected, for example, they will have to download bundles for the internal SPA.

This is an example of how we are thinking the Nuxt project:
* It has hibryd rendering for listing products, navigate categories, and more.
* It has a panel for the customer (customer panel), which will be an SPA for managing orders and tracking shipment.
* It has a SPA panel that will be used for administrative porpuses for company's employees (create products, take orders, etc)

What is your suggestion? Should we keep the nuxt separated for the SPA panel or should we do all in Nuxt?

We actually like the current SPA because it has Pinia and other SPA-related tools that facilitates the administrative tasks.

Thank you!!

**Comment by oneminch** [comment]
I don't if this helps but would Nuxt layers help you with this? 

Nuxt's layers feature allows you to structures your project into multiple separate apps that share some code and configuration like components, composables, etc. It's ideal if code reusability is important to you while running separate applications.
